rust/libceed/src/operator.rs

```diff
@@ -20,18 +20,275 @@
+    ///
+    /// let inputs = op.inputs()?;
+    ///
+    /// assert_eq!(inputs[0].name(), "dx", "Incorrect input name");
+    /// assert_eq!(inputs[1].name(), "weights", "Incorrect input name");
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn name(&self) -> &str {
+        let mut name_ptr: *mut std::os::raw::c_char = std::ptr::null_mut();
+        unsafe {
+            bind_ceed::CeedOperatorFieldGetName(self.ptr, &mut name_ptr);
+        }
+        unsafe { CStr::from_ptr(name_ptr) }.to_str().unwrap()
+    }
+
+    /// Get the ElemRestriction of an OperatorField
+    ///
+    /// ```
+    /// # use libceed::prelude::*;
+    /// # fn main() -> libceed::Result<()> {
+    /// # let ceed = libceed::Ceed::default_init();
+    /// let qf = ceed.q_function_interior_by_name("Mass1DBuild")?;
+    ///
+    /// // Operator field arguments
+    /// let ne = 3;
+    /// let q = 4 as usize;
+    /// let mut ind: Vec<i32> = vec![0; 2 * ne];
+    /// for i in 0..ne {
+    ///     ind[2 * i + 0] = i as i32;
+    ///     ind[2 * i + 1] = (i + 1) as i32;
+    /// }
+    /// let r = ceed.elem_restriction(ne, 2, 1, 1, ne + 1, MemType::Host, &ind)?;
+    /// let strides: [i32; 3] = [1, q as i32, q as i32];
+    /// let rq = ceed.strided_elem_restriction(ne, 2, 1, q * ne, strides)?;
+    ///
+    /// let b = ceed.basis_tensor_H1_Lagrange(1, 1, 2, q, QuadMode::Gauss)?;
+    ///
+    /// // Operator fields
+    /// let op = ceed
+    ///     .operator(&qf, QFunctionOpt::None, QFunctionOpt::None)?
+    ///     .field("dx", &r, &b, VectorOpt::Active)?
+    ///     .field("weights", ElemRestrictionOpt::None, &b, VectorOpt::None)?
+    ///     .field("qdata", &rq, BasisOpt::Collocated, VectorOpt::Active)?;
+    ///
+    /// let inputs = op.inputs()?;
+    ///
+    /// assert!(
+    ///     inputs[0].elem_restriction().is_some(),
+    ///     "Incorrect field ElemRestriction"
+    /// );
+    /// if let ElemRestrictionOpt::Some(r) = inputs[0].elem_restriction() {
+    ///     assert_eq!(
+    ///         r.num_elements(),
+    ///         ne,
+    ///         "Incorrect field ElemRestriction number of elements"
+    ///     );
+    /// }
+    ///
+    /// assert!(
+    ///     inputs[1].elem_restriction().is_none(),
+    ///     "Incorrect field ElemRestriction"
+    /// );
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn elem_restriction(&self) -> ElemRestrictionOpt {
+        let mut ptr = std::ptr::null_mut();
+        unsafe {
+            bind_ceed::CeedOperatorFieldGetElemRestriction(self.ptr, &mut ptr);
+        }
+        if ptr == unsafe { bind_ceed::CEED_ELEMRESTRICTION_NONE } {
+            ElemRestrictionOpt::None
+        } else {
+            let slice = unsafe {
+                std::slice::from_raw_parts(
+                    &ptr as *const bind_ceed::CeedElemRestriction as *const crate::ElemRestriction,
+                    1 as usize,
+                )
+            };
+            ElemRestrictionOpt::Some(&slice[0])
```

Comment:
I found this safe and less contorted way to do this. https://docs.rs/ref-cast/1.0.6/ref_cast/